### PR TITLE
Add global maven goal presets

### PR DESCRIPTION
**Pull request by mbien**
- add `mvnw` updater and a few [maveniverse toolbox](https://github.com/maveniverse/toolbox) diagnostic goals as global preset
 - add right click menu actions which lead to the global and project rule editors for easy add/edit/remove
 - some layout adjustments to both global and custom goal windows so that they resize nicely
 - minor language level cleanup

image

"Edit Global Goals" (new) will open the (already existing) window:
image

(also accessible from maven execution options)

note:
I reverted a few things to not change too much at once. What I originally planned was to combine the global and per-project goal editor windows and I wanted to move the global editor panel to a card in the main maven options. However, the problem is that the two windows and the tools->options windows have different ways how to handle persistence, so this would have been too much at once. Still something we can do at some point.

### CODE REVIEW

**Comment by neilcsmith-net** [comment]
Looks good!

Ideally would have a link to `Edit Project Goals` that opens the right section of the project properties, between the `Other Goals...` and `Edit Global Goals...` links.  Possibly another separator between `Other Goals...` and the edit actions to emphasize execution vs edit?

**Comment by mbien** [comment]
@neilcsmith-net updated the screenshot, let me know what you think

**Comment by neilcsmith-net** [comment]
Thanks!  Think that works and a good change for NB23.

We could perhaps (in future!) standardize some of the terminology here.  We use Goals and Actions in different places, and neither entirely overlaps with Maven's concept of Goal.

**Comment by mbien** [comment]
refreshing since dev build expired

File: java/maven/src/org/netbeans/modules/maven/nbactions.xml [hunk 1]
```diff
@@ -20,4 +20,42 @@
 
 -->
 <actions>
+    <action>
+        <actionName>CUSTOM-Add/Update Wrapper (mvnw)</actionName>
```
[the remark this answers is not in the document]
Comment (mbien) [reply java/maven/src/org/netbeans/modules/maven/nbactions.xml:24]: I considered this, but as soon a user adds/removes/edits them, the defaults _will_ be copied to the user dir.

The action definitions are essentially custom action presets. If the user has a `nbactions.xml` already won't be merged with it.

**Comment by mbien** [comment]
planning to merge this before freeze

**Pull request merged**